x-pack/plugins/triggers_actions_ui/public/application/hooks/use_load_alert_summary.ts

```diff
@@ -17,35 +18,35 @@ export interface AlertSummaryTimeRange {
   title: JSX.Element | string;
 }
 
-interface UseLoadRuleAlertsAggs {
+interface UseLoadAlertSummaryProps {
   features: string;
-  ruleId: string;
   timeRange: AlertSummaryTimeRange;
+  filter?: estypes.QueryDslQueryContainer;
 }
-interface RuleAlertsAggs {
+interface AlertSummary {
   active: number;
   recovered: number;
   error?: string;
 }
 
-interface LoadRuleAlertsAggs {
-  isLoadingRuleAlertsAggs: boolean;
-  ruleAlertsAggs: {
+interface LoadAlertSummary {
+  isLoading: boolean;
+  alertSummary: {
     active: number;
     recovered: number;
   };
-  errorRuleAlertsAggs?: string;
+  error?: string;
 }
 
 interface IndexName {
   index: string;
 }
 
-export function useLoadRuleAlertsAggs({ features, ruleId, timeRange }: UseLoadRuleAlertsAggs) {
+export function useLoadAlertSummary({ features, timeRange, filter }: UseLoadAlertSummaryProps) {
```

Comment:
💭 A lot of changes in this file is just renaming, the most important change is removing `ruleId` and adding `filter` instead.

Reply:
nit - do you think the `timeRange` can be part of `filter` property? maybe something to think in the next PR

Reply:
Interesting idea, Does the API support both fields together? I thought the API needs the time explicitly separate from the rest of the filters to use it in multiple places in the query.